src/dbnode/persist/fs/seek_manager.go

```diff
@@ -437,6 +437,32 @@ func (m *seekerManager) Return(
 	return nil
 }
 
+func (m *seekerManager) ReturnShard(shard uint32) error {
+	if int(shard) >= len(m.seekersByShardIdx) {
+		return errShardNotExists
+	}
```

Comment:
It seems that `seekersByTime()` should be used instead

https://github.com/m3db/m3/blob/255a4b14df1da9b4bc702814933bd8b15d0b3caf/src/dbnode/persist/fs/seek_manager.go#L322-L325

Reply:
`seekersByTime` fails when shard does not exist but we want to close seekers specifically for this use case.